tests/nn/moe/test_moe_layer.py

```diff
@@ -73,6 +73,56 @@ def test_forward(device):
     assert torch.allclose(input, output)
 
 
+# Test Gate which round-robin routes tokens to experts
+class RoundRobinGate(torch.nn.Module):
+    def __init__(self, model_dim, num_experts):
+        super().__init__()
+        self.model_dim = model_dim
+        self.num_experts = num_experts
+
+    def forward(self, input):
+        g, s, _ = input.shape
+        assert s % self.num_experts == 0
+        capacity = 2 * s // self.num_experts
+        output = torch.zeros(g, s, self.num_experts, capacity, dtype=input.dtype, device=input.device)
+        expert = 0
+        c = 0
+        for i in range(s):
+            output[:, i, expert, c] = 1.0
```

Comment:
it seems `expert = i % self.num_experts` and `c = i // self.num_experts`? No need to have separate variables and ++ them?